### PR TITLE
feat(sdk): Upgrade @sentry SDKs to v8.43.0

### PR DESCRIPTION
**Pull request by aliu39**
Planning to instrument the new browser `featureFlagsIntegration` to track internal flag evaluations. We need the latest release for this (https://github.com/getsentry/sentry-javascript/releases/tag/8.43.0).

@sentry/types was deprecated so I've changed imports to use @sentry/core

### CODE REVIEW

**Review by JoshFerge — COMMENTED** [review]
why did import order change?

File: static/app/views/dashboards/widgetBuilder/components/sortBySelector.spec.tsx [hunk 1]
```diff
@@ -110,7 +110,7 @@ describe('WidgetBuilderSortBySelector', function () {
     expect(mockNavigate).toHaveBeenLastCalledWith(
       expect.objectContaining({
         ...router.location,
-        query: expect.objectContaining({sort: ['-count']}),
```
Comment (billyvg) [inline static/app/views/dashboards/widgetBuilder/components/sortBySelector.spec.tsx:113]: Why does this need to be changed?

**Review by billyvg — COMMENTED** [review]
Looks like you need to fix the import in `featureObserver`

**Comment by aliu39** [comment]
> why did import order change?

From my pre-commit, maybe cause it's sorted by module name?

File: static/app/views/dashboards/widgetBuilder/components/sortBySelector.spec.tsx (hunk 1 above)
Comment (aliu39) [reply static/app/views/dashboards/widgetBuilder/components/sortBySelector.spec.tsx:113]: It was failing and `-count()` was the new result. I'm guessing this has to do with an sdk update?

**Comment by billyvg** [comment]
> > why did import order change?
> 
> From my pre-commit, maybe cause it's sorted by module name?

Yeah we had to change imports from `@sentry/types` to `@sentry/core` so it resorted

File: static/app/utils/featureObserver.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import type {FeatureFlagContext} from '@sentry/types/build/types/context';
+import type {FeatureFlagContext} from '@sentry/core/build/types/types-hoist/context';
```
Comment (billyvg) [inline static/app/utils/featureObserver.ts:1]: Should it not be

```suggestion
import type {FeatureFlagContext} from '@sentry/core';
```

Comment (aliu39) [reply static/app/utils/featureObserver.ts:1]: I get an error importing directly. Looks like it's not exported in index

Comment (billyvg) [reply static/app/utils/featureObserver.ts:1]: Let's make sure this gets exported and updated.

**Review by billyvg — APPROVED** [review]

**Pull request merged**